framework/src/outputs/FileOutput.C

```diff
@@ -78,7 +81,7 @@ FileOutput::FileOutput(const InputParameters & parameters) :
   std::string base = "./" + _file_base;
   base = base.substr(0, base.find_last_of('/'));
   if (access(base.c_str(), W_OK) == -1)
-    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
+    mkdir(base.c_str(), S_IRWXU | S_IRGRP);
```

Comment:
@dschwen Good call.  There doesn't seem to be anything like the '-p' option for the mkdir command line utility, so I added a loop to create the intermediate directories if needed.